.travis.yml

```diff
@@ -1,76 +1,176 @@
-sudo: required
-
+---
 # We need the systemd for the kubeadm and it's default from 16.04+
 dist: xenial
+language: minimal
+
+stages:
+  - name: lint
+    if: type IN (push, pull_request)
+  - name: deploy
+    if: type IN (push, pull_request)
+  - name: publish
+    if: type = push AND branch = master
+
+jobs:
+  include:
+    - stage: lint
+      env:
+        - REPO_DIR=/home/travis/build/NBISweden/LocalEGA-helm
+      install:
+        # Installing Helm
+        - curl https://raw.githubusercontent.com/kubernetes/helm/v2.12.2/scripts/get > get_helm.sh && chmod 700 get_helm.sh && ./get_helm.sh
+        - helm init --client-only
+      script:
+        # Check ega-charts format
+        - >
+          for dir in `ls ${REPO_DIR}/ega-charts`; do
+            helm lint ${REPO_DIR}/ega-charts/$dir
+            if [ $? != 0 ]; then
+              travis_terminate 1
+            fi
+          done
+
+    - stage: deploy
+      sudo: required
+      env:
+      # This moves Kubernetes specific config files.
+      # Inspured by https://github.com/LiliC/travis-minikube - Provided under Apache License
+        - CHANGE_MINIKUBE_NONE_USER=true
+      before_script:
+        # Install socat
+        - sudo apt update && sudo apt install socat
+        # Make root mounted as rshared to fix kube-dns issues.
+        - sudo mount --make-rshared /
+        # Download kubectl, which is a requirement for using minikube.
+        - curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.13.0/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
```

Comment:
here we are setting the Kubernetes version to 1.13.0, and we should be good, however we need to think of something like https://docs.travis-ci.com/user/build-matrix/ in order to test the setup on multiple (future, or even past) versions of kubernetes - this is just an idea does not need to be implemented as part of this PR.